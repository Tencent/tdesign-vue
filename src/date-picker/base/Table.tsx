import { defineComponent, PropType, computed } from '@vue/composition-api';
import TDatePickerCell from './Cell';
import { useConfig, usePrefixClass } from '../../hooks/useConfig';
import type { TdDatePickerProps } from '../type';

export default defineComponent({
  name: 'TDatePickerTable',
  props: {
    mode: {
      type: String as PropType<TdDatePickerProps['mode']>,
      default: 'date',
    },
    firstDayOfWeek: Number,
    data: Array,
    time: String,
    onCellClick: Function,
    onCellMouseEnter: Function,
    onCellMouseLeave: Function,
  },
  setup(props) {
    const COMPONENT_NAME = usePrefixClass('date-picker__table');
    const { global } = useConfig('datePicker');
    const { weekdays } = global.value;

    const weekArr = computed(() => {
      const _weekArr = [];
      let wi = props.firstDayOfWeek - 1;
      const len = weekdays.length;
      while (_weekArr.length < len) {
        _weekArr.push(weekdays[wi]);
        wi = (wi + len + 1) % len;
      }

      return _weekArr;
    });

    return { COMPONENT_NAME, weekArr };
  },
  render() {
    const { COMPONENT_NAME, weekArr } = this;

    return (
      <div class={COMPONENT_NAME} onMouseleave={(e: MouseEvent) => this.onCellMouseLeave?.({ e })}>
        <table>
          {this.mode === 'date' && (
            <thead>
              <tr>
                {weekArr.map((value: string, i: number) => (
                  <th key={i}>{value}</th>
                ))}
              </tr>
            </thead>
          )}
          <tbody>
            {this.data.map((row: Array<any>, i: number) => (
              <tr key={i}>
                {row.map((col: any, j: number) => (
                  <TDatePickerCell
                    key={j}
                    {...{
                      props: {
                        time: this.time,
                        onClick: this.onCellClick,
                        onMouseEnter: this.onCellMouseEnter,
                        ...col,
                      },
                    }}
                  />
                ))}
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  },
});
