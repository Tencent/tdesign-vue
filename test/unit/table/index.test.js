import { mount } from '@vue/test-utils';
import Table from '@/src/table/index.ts';

const data = [
  {
    id: 1,
    type: 'Number',
    value: '1',
  },
  {
    id: 2,
    type: 'Number',
    value: '2',
  },
  {
    id: 3,
    type: 'Number',
    value: '3',
  },
];
const columns = [
  {
    title: 'id',
    colKey: 'id',
  },
  {
    title: '类型',
    colKey: 'type',
  },
  {
    title: '取值',
    colKey: 'value',
  },
];
const pagination = {
  pageSize: 10,
  total: 120,
  visibleWithOnePage: true,
};
// every component needs four parts: props/events/slots/functions.
describe('Table', () => {
  // test props api
  describe(':props', () => {
    describe(':prop.data', () => {
      it('`data` is not undefined', () => {
        const wrapper = mount({
          render() {
            return <Table rowKey='id' data={data} columns={columns}></Table>;
          },
        });
        expect(wrapper.vm.$el.getElementsByTagName('tr').length).toBe(data.length + 1);
      });
      it('`data` is undefined', () => {
        const empty = (h) => h({
          template: `<div slot="empty" id="empty-container">
            暂无数据
            </div>`,
        });
        const wrapper = mount({
          render() {
            return (
              <Table rowKey='id' columns={columns} empty={empty}>
              </Table>
            );
          },
        });
        expect(wrapper.find('#empty-container').exists()).toBe(true);
      });
    });
    describe(':props.columns', () => {
      it('`columns` is undefined', () => {
        const wrapper = mount({
          render() {
            return <Table rowKey="id"></Table>;
          },
        });
        expect(wrapper).toMatchSnapshot();
      });
      it('`columns` with children', () => {
        const data = [
          {
            platform: '公有',
            property: 'data',
            type: 'any[]',
            default: '[]',
            needed: 'Y',
            description: '数据源',
          },
          {
            platform: '公有',
            property: 'rowkey',
            type: 'String',
            default: '-1',
            needed: 'Y',
            description: '指定rowkey',
          },
        ];
        const columns = [
          {
            title: '汇总属性',
            children: [
              {
                align: 'left',
                width: '100',
                minWidth: '100',
                className: 'test',
                ellipsis: true,
                colKey: 'platform',
                title: '平台',
              },
              {
                title: '类型及默认值',
                children: [
                  {
                    align: 'left',
                    width: '100',
                    minWidth: '100',
                    className: 'row',
                    ellipsis: true,
                    colKey: 'type',
                    title: '类型',
                  },
                  {
                    align: 'left',
                    width: '100',
                    minWidth: '100',
                    className: 'test4',
                    ellipsis: true,
                    colKey: 'default',
                    title: '默认值',
                  },
                  {
                    align: 'left',
                    width: '100',
                    minWidth: '100',
                    className: 'test3',
                    ellipsis: true,
                    colKey: 'needed',
                    title: '是否必传',
                  },
                ],
              },
            ],
          },
          {
            title: '属性及说明',
            children: [
              {
                align: 'left',
                width: '100',
                minWidth: '100',
                className: 'test2',
                ellipsis: true,
                colKey: 'property',
                title: '属性',
              },
              {
                align: 'left',
                width: '100',
                minWidth: '100',
                className: 'row',
                ellipsis: true,
                colKey: 'description',
                title: '说明',
              },
            ],
          },
        ];
        const wrapper = mount({
          render() {
            return <Table rowKey='id' data={data} columns={columns}></Table>;
          },
        });
        expect(wrapper).toMatchSnapshot();
      });
      it('`columns` with fixed column', () => {
        const columns = [
          {
            title: 'id',
            colKey: 'id',
            fixed: 'left',
          },
          {
            title: '类型',
            colKey: 'type',
          },
          {
            title: '取值',
            colKey: 'value',
          },
        ];
        const wrapper = mount({
          render() {
            return <Table rowKey='id' columns={columns} data={data}></Table>;
          },
        });
        expect(wrapper).toMatchSnapshot();
      });
      it('`columns` with align, width, minWidth and ellipsis', () => {
        const columns = [
          {
            title: 'id',
            colKey: 'id',
            width: 100,
            align: 'left',
            ellipsis: true,
            minWidth: 200,
          },
          {
            title: '类型',
            colKey: 'type',
          },
          {
            title: '取值',
            colKey: 'value',
          },
        ];
        const wrapper = mount({
          render() {
            return <Table rowKey='id' columns={columns} data={data}></Table>;
          },
        });
        expect(wrapper).toMatchSnapshot();
      });
    });
    describe(':props.pagination', () => {
      it('`pagination` visibleWithOnePage is true', () => {
        const wrapper = mount({
          render() {
            return <Table rowKey='id' data={data} columns={columns} pagination={pagination}></Table>;
          },
        });
        expect(wrapper.find('.t-table__pagination').exists()).toBe(true);
      });
    });
    describe(':props.empty', () => {
      it('`empty` is undefined', () => {
        const wrapper = mount({
          render() {
            return <Table rowKey='id' columns={columns}></Table>;
          },
        });
        expect(wrapper.find('.t-table__empty').exists()).toBe(true);
      });
      it('`empty` is a Function', () => {
        const empty = jest.fn(() => {
          /* */
        });
        mount({
          render() {
            return <Table rowKey='id' columns={columns} empty={empty}></Table>;
          },
        });
        expect(empty.mock.calls.length).toBe(1);
      });
    });
    describe(':props.loading', () => {
      it('`loading` is synchronous', () => {
        const loading = true;
        const wrapper = mount({
          render() {
            return <Table rowKey='id' columns={columns} loading={loading}></Table>;
          },
        });
        expect(wrapper.find('.t-loading').exists()).toBe(true);
      });
      it('`loading` is rendered with `slot`', () => {
        const loading = (h) => h({
          template: '<div class="loading__container">loading</div>',
        });
        const wrapper = mount({
          render() {
            return <Table rowKey='id' loading={loading}></Table>;
          },
        });
        expect(wrapper.find('.loading__container').exists()).toBe(true);
      });
      // it('`loading` is asynchronous', () => {
      //   const loading = {
      //     async: true,
      //   };
      //   const wrapper = mount({
      //     render() {
      //       return <Table rowKey='id' columns={columns} loading={loading}></Table>;
      //     },
      //   });
      //   expect(wrapper.find('.t-table--loading-async').exists()).toBe(true);
      // });
    });
    describe(':props.bordered, :props:hover, :props.stripe, :props.size, :props.verticalAlign, :props.height', () => {
      it('size = "small", verticalAlign="middle"', () => {
        const wrapper = mount({
          render() {
            return (
              <Table
                data={data}
                columns={columns}
                bordered={true}
                stripe={true}
                size={'small'}
                hover={true}
                verticalAlign={'middle'}
                rowKey='id'
              ></Table>
            );
          },
        });
        expect(wrapper).toMatchSnapshot();
      });
      it('size = "large", verticalAlign="top"', () => {
        const wrapper = mount({
          render() {
            return <Table rowKey='id' size={'large'} verticalAlign={'top'}></Table>;
          },
        });
        expect(wrapper).toMatchSnapshot();
      });
      it('size = "medium", verticalAlign="bottom"', () => {
        const wrapper = mount({
          render() {
            return <Table rowKey='id' size={'medium'} verticalAlign={'bottom'}></Table>;
          },
        });
        expect(wrapper).toMatchSnapshot();
      });
      it('height = 100', () => {
        const wrapper = mount({
          render() {
            return <Table rowKey='id' height={100} data={data} columns={columns}></Table>;
          },
        });
        expect(wrapper).toMatchSnapshot();
      });
    });
  });

  // test slots
  describe(':slots', () => {
    it('Use slot to customize cell', async () => {
      const slotColumns = columns.slice(0);
      slotColumns[0] = {
        colKey: 'id',
        title: 'id',
        cell: 'idcell',
      };
    });
    it('Use slot to customize empty content', async () => {
      const empty = (h) => h({
        template: '<div class="empty__container">empty</div>',
      });
      const wrapper = await mount({
        render() {
          return <Table rowKey='id' empty={empty} columns={columns}></Table>;
        },
      });
      expect(wrapper.find('.empty__container').exists()).toBe(true);
    });
  });

  // sort
  // describe(':sort', () => {
  //   const data = [
  //     {
  //       key: '1',
  //       firstName: 'Eric',
  //       age: 33,
  //       email: 'espinke0@apache.org',
  //     },
  //     {
  //       key: '2',
  //       firstName: 'Gilberta',
  //       age: 44,
  //       email: 'gpurves1@issuu.com',
  //     },
  //     {
  //       key: '3',
  //       firstName: 'Heriberto',
  //       age: 18,
  //       email: 'hkment2@nsw.gov.au',
  //     },
  //     {
  //       key: '4',
  //       firstName: 'Lazarus',
  //       age: 11,
  //       email: 'lskures3@apache.org',
  //     },
  //     {
  //       key: '5',
  //       firstName: 'Zandra',
  //       age: 22,
  //       email: 'zcroson5@virginia.edu',
  //     },
  //   ];
  //   const columns = [
  //     {
  //       title: 'FirstName',
  //       colKey: 'firstName',
  //     },
  //     {
  //       title: 'Age',
  //       colKey: 'age',
  //       sorter: (a, b) => a.age - b.age,
  //     },
  //     {
  //       title: 'Email',
  //       colKey: 'email',
  //     },
  //   ];
  //   it('click sorter icon', async () => {
  //     const wrapper = await mount({
  //       render() {
  //         return <Table rowKey='id' data={data} columns={columns} />;
  //       },
  //     });
  //     expect(wrapper.find('.table-body tr td').text()).toBe('Eric');
  //     const clickDom = wrapper.find('thead tr td .t-table-sort-asc');
  //     await clickDom.trigger('click');
  //     expect(wrapper.find('.table-body tr td').text()).toBe('Lazarus');
  //     await clickDom.trigger('click');
  //     expect(wrapper.find('.table-body tr td').text()).toBe('Gilberta');
  //   });
  // });

  // asyncLoading
  describe(':asyncLoading', () => {
    // let wrapper;

    // it('init status without async loading row', async () => {
    //   wrapper = await mount({
    //     data() {
    //       return {
    //         asyncLoading: false,
    //       };
    //     },
    //     render() {
    //       const { asyncLoading } = this;
    //       return <Table rowKey='id' data={data} columns={columns} asyncLoading={asyncLoading} />;
    //     },
    //   });
    //   expect(wrapper.find('.table-body tr td .t-table--loading-async').exists()).toBe(false);
    // });

    // it('with async loading row', async () => {
    //   await wrapper.setData({ asyncLoading: true });
    //   expect(wrapper.find('.table-body tr td .t-table--loading-async').exists()).toBe(true);
    // });
  });

  // filters
  describe(':filters', () => {
    let wrapper;
    const data = [
      {
        key: '1',
        firstName: 'Eric',
        lastName: 'Spinke',
        email: 'espinke0@apache.org',
      },
      {
        key: '2',
        firstName: 'Gilberta',
        lastName: 'Purves',
        email: 'gpurves1@issuu.com',
      },
      {
        key: '3',
        firstName: 'Heriberto',
        lastName: 'Kment',
        email: 'hkment2@nsw.gov.au',
      },
      {
        key: '4',
        firstName: 'Lazarus',
        lastName: 'Skures',
        email: 'lskures3@apache.org',
      },
      {
        key: '5',
        firstName: 'Zandra',
        lastName: 'Croson',
        email: 'zcroson5@virginia.edu',
      },
    ];

    it('controlled filters', async () => {
      wrapper = await mount({
        data() {
          return {
            data,
            filteredInfo: null,
          };
        },
        computed: {
          columns() {
            let { filteredInfo } = this;
            filteredInfo = filteredInfo || {};
            const columns = [
              {
                title: 'FirstName',
                colKey: 'firstName',
                key: 'firstName',
                filters: [
                  { label: 'Heriberto', value: 'Heriberto' },
                  { label: 'Eric', value: 'Eric' },
                ],
                filteredValue: filteredInfo.firstName || null,
                onFilter: (value, record) => record.firstName.includes(value),
                filterMultiple: true,
              },
              {
                title: 'LastName',
                colKey: 'lastName',
                key: 'lastName',
              },
              {
                title: 'Email',
                colKey: 'email',
                key: 'email',
                filters: [
                  { label: 'hkment2@nsw.gov.au', value: 'hkment2@nsw.gov.au' },
                  { label: 'lskures3@apache.org', value: 'lskures3@apache.org' },
                ],
                filteredValue: filteredInfo.email || null,
                onFilter: (value, record) => record.email.includes(value),
              },
            ];
            return columns;
          },
        },
        render() {
          const { data, columns } = this;
          return <Table rowKey='id' data={data} columns={columns} />;
        },
      });
      expect(wrapper.findAll('.t-table__body tr').length).toBe(data.length);
      await wrapper.setData({ filteredInfo: { firstName: ['Jim'] } });
      await wrapper.setData({ filteredInfo: null });
      expect(wrapper.findAll('.t-table__body tr').length).toBe(data.length);
    });
  });

  // merge cells
  describe(':rowspanAndColspan', () => {
    it(':rowspanAndColspan', async () => {
      const rowspanAndColspan = ({ col, rowIndex }) => {
        if (col.colKey === 'needed' && rowIndex === 0) {
          return {
            colspan: 2,
          };
        }
        if (col.colKey === 'type' && rowIndex === 1) {
          return {
            colspan: 2,
            rowspan: 2,
          };
        }
      };
      const wrapper = await mount({
        render() {
          return <Table rowKey='id' data={data} columns={columns} rowspanAndColspan={rowspanAndColspan} />;
        },
      });
      const allTr = wrapper.findAll('tr').length;
      const theLastRow = wrapper.findAll('tr').at(allTr - 1);
      expect(theLastRow.findAll('td').length).toBe(1);
    });
  });
});
