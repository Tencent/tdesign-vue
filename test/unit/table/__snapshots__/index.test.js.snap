// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table :props :props.bordered, :props:hover, :props.stripe, :props.size, :props.verticalAlign, :props.height height = 100 1`] = `
<div style="width: 100%;">
  <div class="t-table t-table__header--fixed">
    <div class="t-loading__parent">
      <div class="t-table__content">
        <div class="t-table__header" style="padding-right: 0px;">
          <table style="table-layout: fixed; padding-right: 0px;">
            <colgroup>
              <col>
              <col>
              <col>
            </colgroup>
            <thead columnsprops="[object Object],[object Object],[object Object]">
              <tr>
                <td class="" key="id" colspan="1" rowspan="1">id</td>
                <td class="" key="type" colspan="1" rowspan="1">类型</td>
                <td class="" key="value" colspan="1" rowspan="1">取值</td>
              </tr>
            </thead>
          </table>
        </div>
        <div class="t-table__body" style="height: 100px;">
          <table style="table-layout: fixed;">
            <colgroup>
              <col>
              <col>
              <col>
            </colgroup>
            <tbody class="t-table__body">
              <tr rowkey="id" class="" key="1">
                <td class="" key="id" length="3">1</td>
                <td class="" key="type" length="3">Number</td>
                <td class="" key="value" length="3">1</td>
              </tr>
              <tr rowkey="id" class="" key="2">
                <td class="" key="id" length="3">2</td>
                <td class="" key="type" length="3">Number</td>
                <td class="" key="value" length="3">2</td>
              </tr>
              <tr rowkey="id" class="" key="3">
                <td class="" key="id" length="3">3</td>
                <td class="" key="type" length="3">Number</td>
                <td class="" key="value" length="3">3</td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Table :props :props.bordered, :props:hover, :props.stripe, :props.size, :props.verticalAlign, :props.height size = "large", verticalAlign="top" 1`] = `
<div style="width: 100%;">
  <div class="t-table t-size-l t-table--align-top">
    <div class="t-loading__parent">
      <div class="t-table__content">
        <table style="table-layout: fixed;">
          <colgroup></colgroup>
          <thead columnsprops="">
            <tr></tr>
          </thead>
          <tbody class="t-table__body"></tbody>
        </table>
      </div>
      <div class="t-table__empty">暂无数据</div>
    </div>
  </div>
</div>
`;

exports[`Table :props :props.bordered, :props:hover, :props.stripe, :props.size, :props.verticalAlign, :props.height size = "medium", verticalAlign="bottom" 1`] = `
<div style="width: 100%;">
  <div class="t-table t-table--align-bottom">
    <div class="t-loading__parent">
      <div class="t-table__content">
        <table style="table-layout: fixed;">
          <colgroup></colgroup>
          <thead columnsprops="">
            <tr></tr>
          </thead>
          <tbody class="t-table__body"></tbody>
        </table>
      </div>
      <div class="t-table__empty">暂无数据</div>
    </div>
  </div>
</div>
`;

exports[`Table :props :props.bordered, :props:hover, :props.stripe, :props.size, :props.verticalAlign, :props.height size = "small", verticalAlign="middle" 1`] = `
<div style="width: 100%;">
  <div class="t-table t-table--bordered t-table--striped t-table--hoverable t-size-s">
    <div class="t-loading__parent">
      <div class="t-table__content">
        <table style="table-layout: fixed;">
          <colgroup>
            <col>
            <col>
            <col>
          </colgroup>
          <thead columnsprops="[object Object],[object Object],[object Object]">
            <tr>
              <td class="" key="id" colspan="1" rowspan="1">id</td>
              <td class="" key="type" colspan="1" rowspan="1">类型</td>
              <td class="" key="value" colspan="1" rowspan="1">取值</td>
            </tr>
          </thead>
          <tbody class="t-table__body">
            <tr rowkey="id" class="" key="1">
              <td class="" key="id" length="3">1</td>
              <td class="" key="type" length="3">Number</td>
              <td class="" key="value" length="3">1</td>
            </tr>
            <tr rowkey="id" class="" key="2">
              <td class="" key="id" length="3">2</td>
              <td class="" key="type" length="3">Number</td>
              <td class="" key="value" length="3">2</td>
            </tr>
            <tr rowkey="id" class="" key="3">
              <td class="" key="id" length="3">3</td>
              <td class="" key="type" length="3">Number</td>
              <td class="" key="value" length="3">3</td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Table :props :props.columns \`columns\` is undefined 1`] = `
<div style="width: 100%;">
  <div class="t-table">
    <div class="t-loading__parent">
      <div class="t-table__content">
        <table style="table-layout: fixed;">
          <colgroup></colgroup>
          <thead columnsprops="">
            <tr></tr>
          </thead>
          <tbody class="t-table__body"></tbody>
        </table>
      </div>
      <div class="t-table__empty">暂无数据</div>
    </div>
  </div>
</div>
`;

exports[`Table :props :props.columns \`columns\` with align, width, minWidth and ellipsis 1`] = `
<div style="width: 100%;">
  <div class="t-table">
    <div class="t-loading__parent">
      <div class="t-table__content">
        <table style="table-layout: fixed;">
          <colgroup>
            <col style="width: 100px; min-width: 200px;">
            <col>
            <col>
          </colgroup>
          <thead columnsprops="[object Object],[object Object],[object Object]">
            <tr>
              <td class="t-align-left t-text-ellipsis" key="id" style="overflow: hidden;" colspan="1" rowspan="1">id</td>
              <td class="" key="type" colspan="1" rowspan="1">类型</td>
              <td class="" key="value" colspan="1" rowspan="1">取值</td>
            </tr>
          </thead>
          <tbody class="t-table__body">
            <tr rowkey="id" class="" key="1">
              <td class="t-align-left t-text-ellipsis" key="id" style="overflow: hidden;" length="3">1</td>
              <td class="" key="type" length="3">Number</td>
              <td class="" key="value" length="3">1</td>
            </tr>
            <tr rowkey="id" class="" key="2">
              <td class="t-align-left t-text-ellipsis" key="id" style="overflow: hidden;" length="3">2</td>
              <td class="" key="type" length="3">Number</td>
              <td class="" key="value" length="3">2</td>
            </tr>
            <tr rowkey="id" class="" key="3">
              <td class="t-align-left t-text-ellipsis" key="id" style="overflow: hidden;" length="3">3</td>
              <td class="" key="type" length="3">Number</td>
              <td class="" key="value" length="3">3</td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Table :props :props.columns \`columns\` with children 1`] = `
<div style="width: 100%;">
  <div class="t-table">
    <div class="t-loading__parent">
      <div class="t-table__content">
        <table style="table-layout: fixed;">
          <colgroup>
            <col>
            <col>
          </colgroup>
          <thead columnsprops="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]">
            <tr>
              <td class="" colspan="4" rowspan="1">汇总属性</td>
              <td class="" colspan="2" rowspan="1">属性及说明</td>
            </tr>
            <tr>
              <td class="t-align-left t-text-ellipsis test" key="platform" style="overflow: hidden;" colspan="1" rowspan="2">平台</td>
              <td class="" colspan="3" rowspan="1">类型及默认值</td>
              <td class="t-align-left t-text-ellipsis test2" key="property" style="overflow: hidden;" colspan="1" rowspan="2">属性</td>
              <td class="t-align-left t-text-ellipsis row" key="description" style="overflow: hidden;" colspan="1" rowspan="2">说明</td>
            </tr>
            <tr>
              <td class="t-align-left t-text-ellipsis row" key="type" style="overflow: hidden;" colspan="1" rowspan="1">类型</td>
              <td class="t-align-left t-text-ellipsis test4" key="default" style="overflow: hidden;" colspan="1" rowspan="1">默认值</td>
              <td class="t-align-left t-text-ellipsis test3" key="needed" style="overflow: hidden;" colspan="1" rowspan="1">是否必传</td>
            </tr>
          </thead>
          <tbody class="t-table__body">
            <tr rowkey="id" class="">
              <td class="t-align-left t-text-ellipsis test" key="platform" style="overflow: hidden;" length="6">公有</td>
              <td class="t-align-left t-text-ellipsis row" key="type" style="overflow: hidden;" length="6">any[]</td>
              <td class="t-align-left t-text-ellipsis test4" key="default" style="overflow: hidden;" length="6">[]</td>
              <td class="t-align-left t-text-ellipsis test3" key="needed" style="overflow: hidden;" length="6">Y</td>
              <td class="t-align-left t-text-ellipsis test2" key="property" style="overflow: hidden;" length="6">data</td>
              <td class="t-align-left t-text-ellipsis row" key="description" style="overflow: hidden;" length="6">数据源</td>
            </tr>
            <tr rowkey="id" class="">
              <td class="t-align-left t-text-ellipsis test" key="platform" style="overflow: hidden;" length="6">公有</td>
              <td class="t-align-left t-text-ellipsis row" key="type" style="overflow: hidden;" length="6">String</td>
              <td class="t-align-left t-text-ellipsis test4" key="default" style="overflow: hidden;" length="6">-1</td>
              <td class="t-align-left t-text-ellipsis test3" key="needed" style="overflow: hidden;" length="6">Y</td>
              <td class="t-align-left t-text-ellipsis test2" key="property" style="overflow: hidden;" length="6">rowkey</td>
              <td class="t-align-left t-text-ellipsis row" key="description" style="overflow: hidden;" length="6">指定rowkey</td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Table :props :props.columns \`columns\` with fixed column 1`] = `
<div style="width: 100%;">
  <div class="t-table t-table__cell--fixed t-table--has-fixed">
    <div class="t-loading__parent">
      <div class="t-table__content">
        <table style="table-layout: fixed;">
          <colgroup>
            <col>
            <col>
            <col>
          </colgroup>
          <thead columnsprops="[object Object],[object Object],[object Object]">
            <tr>
              <td class="t-table__cell--fixed-left" key="id" style="position: sticky; left: 0px;" colspan="1" rowspan="1">id</td>
              <td class="" key="type" colspan="1" rowspan="1">类型</td>
              <td class="" key="value" colspan="1" rowspan="1">取值</td>
            </tr>
          </thead>
          <tbody class="t-table__body">
            <tr rowkey="id" class="" key="1">
              <td class="t-table__cell--fixed-left" key="id" style="position: sticky; left: 0px;" length="3">1</td>
              <td class="" key="type" length="3">Number</td>
              <td class="" key="value" length="3">1</td>
            </tr>
            <tr rowkey="id" class="" key="2">
              <td class="t-table__cell--fixed-left" key="id" style="position: sticky; left: 0px;" length="3">2</td>
              <td class="" key="type" length="3">Number</td>
              <td class="" key="value" length="3">2</td>
            </tr>
            <tr rowkey="id" class="" key="3">
              <td class="t-table__cell--fixed-left" key="id" style="position: sticky; left: 0px;" length="3">3</td>
              <td class="" key="type" length="3">Number</td>
              <td class="" key="value" length="3">3</td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
